src/macro/index.ts

```diff
@@ -19,95 +19,91 @@ function camleToDash(string) {
-    }
-    if (newArgs.length > 0) {
-      return t.callExpression(t.identifier('classnames'), newArgs);
-    }
+  
+  function rewriteAndCollectArguments(prefix, argumentPaths, collect) {
+  	return argumentPaths.flatMap((argPath) => {
+          	const node = argPath.node;
+        	if (t.isStringLiteral(node)) {
+             	collect.add(`${prefix}${node.value}`);
+            	return [node]
+            } else if (t.isIdentifier(node)) {
+            	return  [node];
+            } else if (t.isCallExpression(node) && t.isIdentifier(node.callee)) {
+             	// To support import { hover as ho }... 
+            	const name = getClassyName(node.callee.name, argPath.scope);
+              	// When not found it is not part of classy ui
+             	if (name != null) { 
+                  return rewriteAndCollectArguments(`${prefix}${camleToDash(name)}:`, argPath.get('arguments'), collect);
+                }
+            } else if (t.isObjectExpression(node)) {
+            	return node.properties.map((prop) => {
+             		collect.add(`${prefix}${prop.key.value}`);
+                	return t.conditionalExpression(
+                    	prop.value, 
+                    	prop.key,
+                      t.stringLiteral('')
+                    )
+                })           
+            }
+          	return node;
+        });
   }
-
-  function rewriteClasssnames(refs, usedClasses) {
-    refs
-      .map(ref => ref.findParent(p => t.isCallExpression(p)))
-      .forEach(parent => {
-        let stringArgs = extractStringArgs(parent.node.arguments, usedClasses, parent.scope);
-        let objectArgs = extractObjectAndIdentifiersArgs(parent.node.arguments, usedClasses, parent.scope);
-
-        if (stringArgs && objectArgs) {
-          parent.replaceWith(t.binaryExpression('+', objectArgs, stringArgs));
-        } else if (stringArgs) {
-          parent.replaceWith(stringArgs);
-        } else if (objectArgs) {
-          parent.replaceWith(objectArgs);
+	
+
+  const classCollection = new Set();
+  
+  references.classnames
+    .map(ref => ref.findParent(p => t.isCallExpression(p)))
+    .forEach(call => {
+    	const rewrite = rewriteAndCollectArguments('', call.get('arguments'), classCollection);
+    	const newExpression = convertToExpression(rewrite);
+    	if (newExpression) {
+    		call.replaceWith(newExpression);
+        } else {
+          call.remove();
         }
-      });
-  }
-
-  let usedClasses = new Set();
-  rewriteClasssnames(references.classnames, usedClasses);
-
+    });
+    
   const localAddClassUid = state.file.scope.generateUidIdentifier('addClasses');
 
   const runtimeCall = t.callExpression(localAddClassUid, [
     t.arrayExpression(
-      [...usedClasses].reduce((aggr, name) => aggr.concat([t.stringLiteral(name), t.stringLiteral(classes[name])]), []),
+      [...classCollection].reduce((aggr, name) => aggr.concat([t.stringLiteral(name), t.stringLiteral(classes[name] ||'')]), []),
```

Comment:
This line is your extension point. We should see if the classname can be resolve in a separat function and throw a code-frame error so the user an fix it.